custom-local-authenticator/src/main/java/org/wso2/custom/authenticator/local/internal/CustomLocalAuthenticatorServiceComponent.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.custom.authenticator.local.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.custom.authenticator.local.CustomLocalAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.ApplicationAuthenticator;
+import org.wso2.carbon.user.core.service.RealmService;
+
+
+/**
+ * @scr.component name="org.wso2.custom.authenticator.local.basic.component" immediate="true"
+ * @scr.reference name="realm.service"
+ * interface="org.wso2.carbon.user.core.service.RealmService"cardinality="1..1"
+ * policy="dynamic" bind="setRealmService" unbind="unsetRealmService"
+ */
+public class CustomLocalAuthenticatorServiceComponent {
```

Comment:
Shall we change to the  SCR annotations?

Reply:
Change into what?

Reply:
Instead of defining in the java doc, we can use the annotation based OSGi service registration. Better to do that as java doc based approach is not used anymore